v3/src/components/graph/hooks/use-adornment-attributes.ts

```diff
@@ -1,20 +1,30 @@
+import { useRef } from "react"
 import { ScaleNumericBaseType } from "../../axis/axis-types"
 import { useGraphDataConfigurationContext } from "./use-graph-data-configuration-context"
 import { useGraphLayoutContext } from "./use-graph-layout-context"
+import { useGraphContentModelContext } from "./use-graph-content-model-context"
 
 export const useAdornmentAttributes = () => {
+  const graphModel = useGraphContentModelContext()
   const dataConfig = useGraphDataConfigurationContext()
   const layout = useGraphLayoutContext()
+  const adornmentsStore = graphModel.adornmentsStore
   const xAttrType = dataConfig?.attributeType("x")
   const xScale = layout.getAxisScale("bottom") as ScaleNumericBaseType
   const yScale = layout.getAxisScale("left") as ScaleNumericBaseType
   const allAttributes = dataConfig?.dataset?.attributes
   const xAttrId = dataConfig?.attributeID("x") || ""
   const yAttrId = dataConfig?.attributeID("y") || ""
+  const numericAttrId = xAttrId && xAttrType === "numeric" ? xAttrId : yAttrId
```

Comment:
Good question. I think it's fine since the non-univariate adornments don't use it and in the future any caller who did would be warned by the property name.